### DISCUSSION TITLE
Scaling variables in manual_backward when using mixed precision

### DISCUSSION THREAD

**Discussion by marcmk6** [Lightning-AI/pytorch-lightning#13424] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule]
Hi,

The algorithm that I'm working on directly issues gradient of "loss" w.r.t. model predictions, so I need get related variables (`model_prediction` and `gradient` in the following example) scaled when using mixed precision. How can I achieve this?

```python
def training_step(...):
    opt = self.optimizers()
    loss = torch.tensor(0) # not used in my case
    opt.zero_grad()
    # automatically applies scaling, etc...
    self.manual_backward(loss, {'model_prediction': model_prediction, 'grad': gradient})
    opt.step()

def backward(loss, optimizer, optimizer_idx, *args, **kwargs):
    # `loss` is not used
    prediction = kwargs['model_prediction']
    grad = kwargs['grad']
    # how to get these two variables scaled? 
    prediction.backward(grad)
```

Thanks

**Comment by marcmk6** [comment]
@rohitgr7 Could you help with this?
